graylog2-web-interface/src/components/common/SourceCodeEditor.jsx

```diff
@@ -0,0 +1,205 @@
+    width: PropTypes.number,
+  }
+
+  static defaultProps = {
+    annotations: [],
+    focus: false,
+    fontSize: 13,
+    height: 200,
+    mode: 'text',
+    onChange: () => {},
+    onLoad: () => {},
+    readOnly: false,
+    resizable: true,
+    theme: 'light',
+    toolbar: true,
+    value: '',
+    width: Infinity,
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      height: props.height,
+      width: props.width,
+      selectedText: '',
+    };
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.height !== prevProps.height || this.props.width !== prevProps.width) {
+      this.reloadEditor();
+    }
+  }
+
+  handleResize = (event, { size }) => {
+    const { height, width } = size;
+    this.setState({ height: height, width: width }, this.reloadEditor);
+  }
+
+  reloadEditor = () => {
+    if (this.props.resizable) {
+      this.reactAce.editor.resize();
+    }
+  }
+
+  isCopyDisabled = () => this.props.readOnly || this.state.selectedText === '';
+
+  isPasteDisabled = () => this.props.readOnly;
+
+  isRedoDisabled = () => this.props.readOnly || !this.reactAce || !this.reactAce.editor.getSession().getUndoManager().hasRedo();
+
+  isUndoDisabled = () => this.props.readOnly || !this.reactAce || !this.reactAce.editor.getSession().getUndoManager().hasUndo();
+
+  handleRedo = () => {
+    this.reactAce.editor.redo();
+    this.focusEditor();
+  }
+
+  handleUndo = () => {
+    this.reactAce.editor.undo();
+    this.focusEditor();
+  }
+
+  handleSelectionChange = (selection) => {
+    if (!this.reactAce || !this.props.toolbar || this.props.readOnly) {
+      return;
+    }
+    const selectedText = this.reactAce.editor.getSession().getTextRange(selection.getRange());
+    this.setState({ selectedText: selectedText });
+  }
+
+  focusEditor = () => {
+    this.reactAce.editor.focus();
+  }
+
+  render() {
+    const { height, width } = this.state;
+    const { theme, resizable } = this.props;
+    const validCssWidth = Number.isNaN(width) ? '100%' : width;
+    const containerStyle = `${style.sourceCodeEditor} ${theme !== 'light' && style.darkMode} ${!resizable && style.static}`;
+    const overlay = <Tooltip id={'paste-button-tooltip'}>Press Ctrl+V (&#8984;V in macOS) or select Edit&thinsp;&rarr;&thinsp;Paste to paste from clipboard.</Tooltip>;
+    return (
+      <div>
+        {this.props.toolbar &&
+          <div className={style.toolbar} style={{ width: validCssWidth }}>
+            <ButtonToolbar>
+              <ButtonGroup>
+                <ClipboardButton title={<i className="fa fa-copy fa-fw" />}
+                                 bsStyle="link"
+                                 bsSize="sm"
+                                 onSuccess={this.focusEditor}
+                                 text={this.state.selectedText}
+                                 buttonTitle="Copy (Ctrl+C / &#8984;C)"
+                                 disabled={this.isCopyDisabled} />
```

Comment:
This passes the `isCopyDisabled` function instead of executing it.

Reply:
Good finding! This got broken in one of the changes I did.